client/grandpa-ranger/src/util.ts

```diff
@@ -1,85 +1,102 @@
-import { tmpdir } from "os"
-import { join } from "path"
-import { writeFile } from "fs/promises"
-import { promisify } from "util"
-import { exec as _exec } from "child_process"
 import { TypeRegistry, createType } from "@polkadot/types"
 import { Header } from "@polkadot/types/interfaces"
 import { ApiPromise } from "@polkadot/api"
 import { BN } from "@polkadot/util"
 
 const registry = new TypeRegistry()
+const justification = { type: 'GrandpaJustification<Header>' }
+export const decodeJustificationTarget = (data: any) => {
+    const justification = decodeJustification(data)
 
-export const decodeCustomType = (type: string, data: string) => {
-  const typeObject = { type }
-  registry.register(typeObject)
-  return createType(registry, typeObject.type, data.trim())
+    return justification.commit.targetNumber.toNumber()
 }
 
-export const exec = promisify(_exec)
+// gets events of a specific block, and checks if authoritySetUpdate is included.
+// it would be more efficient to listen for the NewAuthorities event directly, but I'm unable to decode the block number currently.
+export const containsAuthoritySetUpdate= async (api: any, blockNumber: number) => {
+    const hash = await api.rpc.chain.getBlockHash(blockNumber);
+    const notifications = await api.query.system.events.at(hash);
+    return notifications.some(noti => noti.event.method === "NewAuthorities")
+}
```

Comment:
very inefficient way of checking if a justification contains an authority set update